File: BooksSeller.WebApi/BooksSeller.WebApi/Models/Book.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.Serialization;
using System.Web;

namespace BooksSeller.WebApi.Models
{
    [DataContract]
    public class Book
    {
        [DataMember]
        public string Code { get; set; }        [DataMember]
        public string Title { get; set; }        [DataMember]
        public DateTime ReleaseDate { get; set; }        [DataMember]
        public double Price { get; set; }
    }
}

File: web/app/src/typescript/models/Book.ts
/// <reference path="../references.ts"/>

module Books.Seller.Web.Models {
  export class Book {
    public Code : string;
    public Title : string;
    public ReleaseDate : Date;
    public Price : number;

    constructor(code: string, title: string, availability: Date) {
      this.Code = code;
      this.Title = title;
      this.ReleaseDate = availability;
      this.Price = 0.0;
    }
  }
}
